macros/src/main/scala-3/com/softwaremill/macwire/internals/EligibleValuesFinder.scala

```diff
@@ -93,6 +114,20 @@ private[macwire] class EligibleValuesFinder[Q <: Quotes](log: Logger)(using val
     doFind(Symbol.spliceOwner, Scope.init)
   }
 
+  private def notJunk(member: Symbol): Boolean = {
+    !member.fullName.startsWith("java.lang.Object") &&
+    !member.fullName.startsWith("scala.Any") &&
+    !member.fullName.startsWith("scala.AnyRef") &&
+    !member.fullName.endsWith("<init>") &&
+    !member.fullName.endsWith("$init$") &&
+    !member.fullName.contains("$default$") &&
+    !member.fullName.matches(".*_\\d+")
```

Comment:
why is this junk? :)

Reply:
not sure about inits, but the other two are generated on scala3 for case classes and break resolution

Reply:
sure `$init$` and `$default$` I understand, but the last one seems to be very generic, and either needs a comment (what cases it covers) or maybe we can get rid of it?